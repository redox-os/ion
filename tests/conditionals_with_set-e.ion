set -e

true && echo and || echo or
false && echo and || echo or
false || echo false
true && echo true
true || echo cant get here
false || false || echo double or
true && true && echo double and
false || true && echo or and
contains "one two three" two && echo true || echo false
contains "one two three" abc two && echo true || echo false
contains "one two three" foo bar && echo true || echo false
starts-with "one two three" one && echo true || echo false
starts-with "one two three" abc one && echo true || echo false
starts-with "one two three" three && echo true || echo false
ends-with "one two three" three && echo true || echo false
ends-with "one two three" abc three && echo true || echo false
ends-with "one two three" one two && echo true || echo false

false && echo cant get here