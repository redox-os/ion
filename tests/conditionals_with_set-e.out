and
or
false
true
double or
double and
or and
true
true
false
true
true
false
true
true
false
ion: pipeline execution error: early exit: pipeline failed with error code 1
